datalad/support/gitrepo.py

```diff
@@ -671,21 +674,174 @@ def git_remote_update(self, name='', verbose=False):
+        if remote is None:
+            if refspec is not None:
+                # conflicts with using tracking branch or fetch all remotes
+                # For now: Just fail.
+                # TODO: May be check whether it fits to tracking branch
+                raise ValueError("refspec specified without a remote. (%s)" %
+                                  refspec)
+            if all:
+                remotes_to_push = self.repo.remotes
+            else:
+                # No explicit remote to fetch.
+                # => get tracking branch:
+                tb = self.repo.active_branch.tracking_branch().name
+                if tb:
+                    parts = tb.split('/')
+                    assert len(parts) == 2
+                    remotes_to_push = [self.repo.remote(parts[0])]
+                    refspec = parts[1]
+                else:
+                    # No remote, no tracking branch
+                    # => fail
+                    raise ValueError("No remote specified to fetch from nor a "
+                                     "tracking branch is set up.")
+        else:
+            remotes_to_push = [self.repo.remote(remote)]
+
+        for rm in remotes_to_push:
+            push_url = \
+                rm.config_reader.get('pushurl'
+                                     if rm.config_reader.has_option('pushurl')
+                                     else 'url')
+            if push_url.startswith('ssh:'):
+                cnct = ssh_manager.get_connection(push_url)
+                cnct.open()
+                # TODO: with git <= 2.3 keep old mechanism:
+                #       with rm.repo.git.custom_environment(GIT_SSH="wrapper_script"):
+                with rm.repo.git.custom_environment(
+                        GIT_SSH_COMMAND="ssh -S %s" % cnct.ctrl_path):
+                    rm.push(refspec=refspec, progress=progress, **kwargs)
+                    # TODO: progress +kwargs
+            else:
+                rm.push(refspec=refspec, progress=progress, **kwargs)
+                # TODO: progress +kwargs
 
     def git_get_remote_url(self, name, push=False):
-        """We need to know, where to clone from, if a remote is
-        requested
-        """
+        """Get the url of a remote.
 
-        remote = self.repo.remote(name)
-        return remote.config_reader.get(
-            'pushurl' if push and remote.config_reader.has_option('pushurl') else 'url')
+        Reads the configuration of remote `name` and returns its url or None,
+        if there is no url configured.
+
+        Parameters
+        ----------
+        name: str
+          name of the remote
+        push: bool
+          if True, get the pushurl instead of the fetch url.
+        """
+        cfg_reader = self.repo.remote(name).config_reader
+        if push:
+            if cfg_reader.has_option('pushurl'):
+                return cfg_reader.get('pushurl')
+            else:
+                return None
```

Comment:
just expressing myself:  silly of GitPython to not provide proper "proxying" of the default kwarg in this 'partial' (I think)